### PR TITLE
added FileMutator 

### PR DESCRIPTION
**Pull request by dg**
- new feature? yes
- BC break? no

This can be used mocking of final classes/members and also as foundation for #17

### CODE REVIEW

File: src/Framework/FileMutator.php [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+	public function stream_open($path, $mode, $options, &$openedPath)
+	{
+		$usePath = (bool) ($options & STREAM_USE_PATH);
+		if (pathinfo($path, PATHINFO_EXTENSION) === 'php') {
+			$content = $this->native('file_get_contents', $path, $usePath, $this->context);
+			if ($content === FALSE) {
+				return FALSE;
+			} else {
+				$this->handle = tmpfile();
```
Comment (milo) [inline src/Framework/FileMutator.php]: It opens file in `w+` mode, not sure it's important or not.

**Comment by fprochazka** [comment]
image

What a lovely hack (no sarcasm). Muggles will enjoy using this to hack libraries with final in their applications :)

**Comment by JanTvrdik** [comment]
Should I write to php-internals mailing list suggestion to disable this?
I still feel like this is too evil to be allowed to exist.

**Comment by f3l1x** [comment]
Thats amazing feature. Can't wait.

**Comment by milo** [comment]
Merged 8c9d656b6111d1c40a90378264e0de44e1fa72e0. Let's experiment.

**Comment by dg** [comment]
That's a bit premature, I do not like current API.

**Comment by milo** [comment]
Oh, sorry. I removed it from master. I misunderstood you on PoSobota.

**Comment by dg** [comment]
From my point this is ready to merge.

**Comment by milo** [comment]
👍

**Pull request merged**